### DISCUSSION TITLE
What is the streaming protocol used in Weylus?

### DISCUSSION THREAD

**Discussion by KarimElsayad247** [H-M-H/Weylus#305] [Q&A] [answered]
Does it stream over HTTP or RTSP? is it a standard protocol or a custom made one? I'm currently looking for a library to stream Weylus on android

https://developer.android.com/media/platform/supported-formats#network

**Comment by KarimElsayad247** [comment]
BTW it felt like this is the function that throws the video packet over the wire, but not really sure

```rust
// this is used as callback in lib/encode_video.c via ffmpegs AVIOContext
#[no_mangle]
fn write_video_packet(video_encoder: *mut c_void, buf: *const c_uchar, buf_size: c_int) -> c_int {
    let video_encoder = unsafe { (video_encoder as *mut VideoEncoder).as_mut().unwrap() };
    (video_encoder.write_data)(unsafe {
        std::slice::from_raw_parts(buf as *const u8, buf_size as usize)
    });
    0
}
```

**Comment by H-M-H** [comment, marked as answer]
Weylus sends fragmented mp4 frames with one video stream encoded using the H.264 codec as binary messages over the websocket connection. See https://github.com/H-M-H/Weylus/blob/d30b8b3e56820b72a858e227654823722f0d5d8f/ts/lib.ts#L946-L960 and https://github.com/H-M-H/Weylus/blob/d30b8b3e56820b72a858e227654823722f0d5d8f/ts/lib.ts#L917-L941 for how it is handled in a browser based client.
